edgelet/iotedge/src/init/import/old_config/mod.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+mod agent;
+pub(crate) use agent::*;
+
+mod moby_runtime;
+pub(crate) use moby_runtime::*;
+
+mod provisioning;
+pub(crate) use provisioning::*;
+
+use std::path::PathBuf;
+
+use url::Url;
+
+#[derive(Debug, serde_derive::Deserialize)]
+pub(crate) struct Config {
+    pub(crate) provisioning: Provisioning,
+
+    pub(crate) agent: ModuleSpec,
+
+    pub(crate) hostname: String,
+    pub(crate) parent_hostname: Option<String>,
+
+    pub(crate) connect: Connect,
+    pub(crate) listen: Listen,
+
+    pub(crate) homedir: PathBuf,
+
+    pub(crate) certificates: Option<Certificates>,
+
+    #[serde(default)]
+    pub(crate) watchdog: WatchdogSettings,
+
+    pub(crate) moby_runtime: MobyRuntime,
+}
+
+#[derive(Debug, serde_derive::Deserialize)]
+pub(crate) struct Connect {
+    workload_uri: Url,
+    management_uri: Url,
+}
+
+#[derive(Debug, serde_derive::Deserialize)]
+pub(crate) struct Listen {
+    workload_uri: Url,
+    management_uri: Url,
+}
+
+#[derive(Debug, serde_derive::Deserialize)]
+pub(crate) struct Certificates {
+    #[serde(flatten)]
+    pub(crate) device_cert: Option<DeviceCertificate>,
+
+    #[serde(default = "default_auto_generated_ca_lifetime_days")]
+    pub(crate) auto_generated_ca_lifetime_days: u16,
+}
+
+#[derive(Debug, serde_derive::Deserialize)]
+pub(crate) struct DeviceCertificate {
+    pub(crate) device_ca_cert: String,
+    pub(crate) device_ca_pk: String,
+    pub(crate) trusted_ca_certs: String,
+}
+
+fn default_auto_generated_ca_lifetime_days() -> u16 {
```

Comment:
`const DEFAULT_AUTO_GENERATED_CA_LIFETIME_DAYS: u16 = 90`?

or at the very least, make this a `const fn`

Reply:
`#[serde(default)]` needs a path that can be invoked as a function.

Reply:
Done.